device_plugins/sfc_nic/sfc_nic_plugin.go

```diff
@@ -0,0 +1,474 @@
+		cmdName = "onload_tool unload"
+		out, err = ExecCommand("onload_tool", "unload")
+		//fmt.Println("CMD--" + cmdName + ": " + out.String())
+		cmdName = "onload_uninstall"
+		out, err = ExecCommand(cmdName)
+		//fmt.Println("CMD--" + cmdName + ": " + out.String())
+
+		os.Chdir(os.Getenv("HOME"))
+		// remove current onload
+		cmdName = "rm onload"
+		out, err = ExecCommand("/bin/sh", "-c", "rm -rf ./openonload*")
+		//fmt.Println("CMD--" + cmdName + ": " + out.String())
+
+		os.Chdir(os.Getenv("HOME"))
+		// get open onload from a authorized source - further security todo
+		cmdName = "get onload"
+
+		if strings.HasPrefix(onloadsrc, "http://") {
+			out, err = ExecCommand("wget", onloadsrc)
+		} else {
+			out, err = ExecCommand("cp", onloadsrc, ".")
+		}
+		//fmt.Println("CMD--" + cmdName + ": " + out.String())
+
+		os.Chdir(os.Getenv("HOME"))
+		// unzip onload
+		cmdName = "unzip onload"
+		cmdstring := "./openonload-" + onloadver + ".tgz"
+		out, err = ExecCommand("tar", "xvzf", cmdstring)
+		//fmt.Println("CMD--" + cmdName + ": " + out.String())
+
+		os.Chdir(os.Getenv("HOME"))
+		// install current onload
+		cmdName = "./openonload-" + onloadver + "/scripts/onload_install"
+		out, err = ExecCommand(cmdName)
+		if err != nil {
+			return fmt.Errorf("onload_install failed: %v", err)
+		}
+		if strings.Contains(out.String(), "onload_install: Install complete") {
+			fmt.Println("CMD--" + cmdName + ": " + "Install complete")
+
+			// reload onload
+			cmdName = "onload_tool unload"
+			_, err = ExecCommand("onload_tool", "unload")
+			//fmt.Println("CMD--" + cmdName + ": " + out.String())
+			cmdName = "onload_tool reload"
+			_, err = ExecCommand("onload_tool", "reload")
+			if !sfc.isOnloadInstallHealthy() {
+				return fmt.Errorf("Onload Install Failed!!")
+			}
+		} else {
+			return fmt.Errorf("onload_install could not be completed!!!!")
+		}
+	} else {
+		return err
+	}
+	return nil
+}
+
+func (sfc *sfcNICManager) Init() error {
+	glog.Info("Init\n")
+	err := sfc.installOnload()
+	return err
+}
+
+func Register(kubeletEndpoint string, pluginEndpoint, socketName string) error {
+	conn, err := grpc.Dial(kubeletEndpoint, grpc.WithInsecure(),
+		grpc.WithDialer(func(addr string, timeout time.Duration) (net.Conn, error) {
+			return net.DialTimeout("unix", addr, timeout)
+		}))
+	defer conn.Close()
+	if err != nil {
+		return fmt.Errorf("device-plugin: cannot connect to kubelet service: %v", err)
+	}
+	client := pluginapi.NewRegistrationClient(conn)
+	reqt := &pluginapi.RegisterRequest{
+		Version:      pluginapi.Version,
+		Endpoint:     pluginEndpoint,
+		ResourceName: resourceName,
+	}
+
+	_, err = client.Register(context.Background(), reqt)
+	if err != nil {
+		return fmt.Errorf("device-plugin: cannot register to kubelet service: %v", err)
+	}
+	return nil
+}
+
+// Implements DevicePlugin service functions
+func (sfc *sfcNICManager) ListAndWatch(emtpy *pluginapi.Empty, stream pluginapi.DevicePlugin_ListAndWatchServer) error {
```

Comment:
Not advance Go lang dev here, but should this actually looks like `func (d *DevicePluginServer) ListAndWatch` to let Go know that this implement the service interface in the `pluginapi` ?